overlord/configstate/configcore/services.go

```diff
@@ -231,6 +236,87 @@ func handleServiceDisableConfiguration(_ sysconfig.Device, tr config.ConfGetter,
 			}
 		}
 	}
+	// configure ssh ports
+	if err := handleServiceConfigSSHPort(dev, tr, opts); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func validateServiceConfiguration(tr config.ConfGetter) error {
+	// validate the ssh port setting
+	output, err := coreCfg(tr, sshPortOpt)
+	if err != nil {
+		return err
+	}
+	if output == "" {
+		return nil
+	}
+	port, err := strconv.Atoi(output)
+	if err != nil {
+		return err
+	}
+	if port > 65535 || port < 1 {
+		return fmt.Errorf("cannot use port %v: must be in the range 1-65535", port)
+	}
+	return nil
+}
+
+func handleServiceConfigSSHPort(dev sysconfig.Device, tr config.ConfGetter, opts *fsOnlyContext) error {
+	// see if anything needs to happenhan
+	var pristineSSHPort, newSSHPort interface{}
+
+	if err := tr.GetPristine("core", sshPortOpt, &pristineSSHPort); err != nil && !config.IsNoOption(err) {
+		return err
+	}
+	if err := tr.Get("core", sshPortOpt, &newSSHPort); err != nil && !config.IsNoOption(err) {
+		return err
+	}
+	if pristineSSHPort == newSSHPort {
+		return nil
+	}
+
+	// ssh.port config has changed, write new config
+	root := dirs.GlobalRootDir
+	if opts != nil {
+		root = opts.RootDir
+	}
+
+	// Note: Only UC20+ supports using the "sshd_config.d"
+	// dir. Supporting older systems would be hard because we
+	// would have to merge somehow the UC16 and UC18 config in a
+	// fsOnlyContext
+	if !dev.HasModeenv() {
+		return fmt.Errorf("cannot set ssh port configuration on systems older than UC20")
+	}
+	name := "port.conf"
+	dirContent := map[string]osutil.FileState{}
+	if newSSHPort != nil && newSSHPort != "" {
+		dirContent[name] = &osutil.MemoryFileState{
+			Content: []byte(fmt.Sprintf("Port %v\n", newSSHPort)),
```

Comment:
as I mentioned in the sprint I wonder if we should instead support the full ListenAddress instead of just port

Reply:
I like that! `sshd` supports multiple listen addresses, I guess we can just support them as a comma separated list? Would it still be: `snap set system server.ssh.port=localhost:80,hostname:8022` or would you prefer something other then `ssh.port` (like `ssh.listen` or `ssh.listen-addresss`)?

Reply:
listen-address, comma separated is fine.  Reading the docs we need to double check if `ListenAddress :port` is equivalent to `Port port`?  also it seems ListenAddress doesn't support omitting ":" in this case, but maybe we should?

Reply:
I looked at the openssh code now and also experimented a bit. It seems like `ListenAddress :port` does not work, it always needs a hostname or ip. So we have two options (it seems):
1. if only a port is specified just use `Port <port>`
2. if only a port is specific, generate something like: `ListenAddress 0.0.0.0:<port>\nListenAddress [::1]:<port>`

We also need to think if the option should be `service.ssh.listen` instead maybe? Or we provide `.port` and `.listen` but that feels a bit redundant

Reply:
maybe 2. is more annoying but easier to understand/predict?  

regarding the name I think mentioned `listen-address`in the previous comment, most option names are names not verbs

Reply:
I looked into this now again and will implement it using `listen-address`. However I would like not to support "naked"  ports because it ambitious, i.e. `service.ssh.listen-address=2001` could mean the port `2001` but there might also be a hostname `2001` on the network. So I think it's either:
1. service.ssh.port and service.ssh.listen-address as separate options if we want to support "naked" ports
2. service.ssh.listen-address for port only should specify `:8022`
Or am I missing something?

Reply:
I don't have a strong opinion, I would think 2. is ok as long as we document things properly